handlers/validator.go

```diff
@@ -595,34 +595,35 @@ func Validator(w http.ResponseWriter, r *http.Request) {
 		}
 
 		if validatorPageData.AttestationsCount > 0 {
-			// get attestationStats from validator_stats
-			attestationStats := struct {
-				MissedAttestations uint64 `db:"missed_attestations"`
-			}{}
-			if lastStatsDay > 0 {
-				err := db.ReaderDb.Get(&attestationStats, "SELECT missed_attestations_total AS missed_attestations FROM validator_stats WHERE validatorindex = $1 AND day = $2", index, lastStatsDay)
-				if err == sql.ErrNoRows {
-					logger.Warningf("no entry in validator_stats for validator index %v while lastStatsDay = %v", index, lastStatsDay)
-				} else if err != nil {
-					return fmt.Errorf("error getting validator attestationStats while lastStatsDay = %v: %w", lastStatsDay, err)
-				}
-			}
-
-			// add attestationStats that are not yet in validator_stats
-			lookback := int64(lastFinalizedEpoch - (lastStatsDay+1)*utils.EpochsPerDay())
-			if lookback > 0 {
-				missedAttestations, err := db.BigtableClient.GetValidatorMissedAttestationHistory([]uint64{index}, lastFinalizedEpoch-uint64(lookback), lastFinalizedEpoch)
-				if err != nil {
-					return fmt.Errorf("error getting validator attestations not in stats from bigtable: %w", err)
-				}
-				attestationStats.MissedAttestations += uint64(len(missedAttestations[index]))
-			}
-
 			if isPreGenesis {
 				validatorPageData.MissedAttestationsCount = 0
 				validatorPageData.ExecutedAttestationsCount = 0
 				validatorPageData.UnmissedAttestationsPercentage = 1
 			} else {
+				// get attestationStats from validator_stats
+				attestationStats := struct {
+					MissedAttestations uint64 `db:"missed_attestations"`
+				}{}
+				if lastStatsDay > 0 {
+					err := db.ReaderDb.Get(&attestationStats, "SELECT missed_attestations_total AS missed_attestations FROM validator_stats WHERE validatorindex = $1 AND day = $2", index, lastStatsDay)
+					if err == sql.ErrNoRows {
+						logger.Warningf("no entry in validator_stats for validator index %v while lastStatsDay = %v", index, lastStatsDay)
+					} else if err != nil {
+						return fmt.Errorf("error getting validator attestationStats while lastStatsDay = %v: %w", lastStatsDay, err)
+					}
+				}
+
+				// add attestationStats that are not yet in validator_stats (if any)
+				nextStatsDayFirstEpoch, _ := utils.GetFirstAndLastEpochForDay(lastStatsDay + 1)
+				if validatorPageData.Epoch > nextStatsDayFirstEpoch {
+					lookback := uint64(validatorPageData.Epoch - nextStatsDayFirstEpoch)
```

Comment:
nitpick:
```suggestion
					lookback := validatorPageData.Epoch - nextStatsDayFirstEpoch
```
Cast in line 620 also unnecessary.

Reply:
Done.